test/NuGet.Services.Staging.EndToEndTests/StagingClient/StagingClient.cs

```diff
@@ -0,0 +1,280 @@
+
+            Func<HttpRequestMessage> requestFactory = () =>
+            {
+
+                var request = new HttpRequestMessage(HttpMethod.Post, new Uri(_stagingServiceUri, $"{StagePath}/{stageId}"));
+                request.Headers.Add(ApiKeyHeader, apiKey);
+
+                return request;
+            };
+
+            await SendAsync(requestFactory);
+        }
+
+        public async Task<JObject> GetCommitProgress(string stageId)
+        {
+            _logger.WriteLine($"StagingClient: GetCommitProgress called for stage {stageId}");
+
+            Func<HttpRequestMessage> requestFactory = () =>
+            {
+                var request = new HttpRequestMessage(HttpMethod.Get, new Uri(_stagingServiceUri, $"{StagePath}/{stageId}/commit"));
+                return request;
+            };
+
+            var responseBody = await SendAsync(requestFactory);
+
+            return JObject.Parse(responseBody);
+        }
+
+        public async Task<JObject> Index(string stageId)
+        {
+            _logger.WriteLine($"StagingClient: Index called for stage {stageId}");
+
+            Func<HttpRequestMessage> requestFactory = () =>
+            {
+                var request = new HttpRequestMessage(HttpMethod.Get, new Uri(_stagingServiceUri, $"{StagePath}/{stageId}/index.json"));
+                return request;
+            };
+
+            var responseBody = await SendAsync(requestFactory);
+
+            return JObject.Parse(responseBody);
+        }
+
+        public async Task<JObject> Query(string stageId, string queryString)
+        {
+            _logger.WriteLine($"StagingClient: Query called for stage {stageId}");
+
+            JObject index = await Index(stageId);
+
+            string searchEndpoint = index["resources"].Where(x => x["@type"].ToString() == ServiceTypes.SearchQueryService[0]).ToString();
+
+            Func<HttpRequestMessage> requestFactory = () =>
+            {
+                var request = new HttpRequestMessage(HttpMethod.Get, new Uri(_stagingServiceUri, $"{searchEndpoint}?q={queryString}"));
+                return request;
+            };
+
+            var responseBody = await SendAsync(requestFactory);
+
+            return JObject.Parse(responseBody);
+        }
+
+        public async Task PushPackage(string stageId, string apiKey, Stream packageStream)
+        {
+            _logger.WriteLine($"StagingClient: PushPackage called for stage {stageId}");
+
+            JObject index = await Index(stageId);
+
+            string pushEndpoint = index["resources"].First(x => x["@type"].ToString() == ServiceTypes.PackagePublish)["@id"].ToString();
+
+            var content = new MultipartFormDataContent();
+            packageStream.Position = 0;
+            var packageContent = new StreamContent(packageStream);
+            packageContent.Headers.ContentType = MediaTypeHeaderValue.Parse("application/octet-stream");

```

Comment:
fixed

---
In reply to: [64174075](https://github.com/NuGet/NuGet.Staging/pull/49#discussion_r64174075) [](ancestors = 64174075)